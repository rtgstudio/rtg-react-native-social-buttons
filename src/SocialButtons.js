import React, { Component } from "react";
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Image,
  Alert
} from "react-native";

export default class SocialButton extends Component {
  constructor(props) {
    super(props);
  }

  renderFaceBookButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.FacebookStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/facebook.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  renderGoogleButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.GooglePlusStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/google-plus.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  renderGithubButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.GithubStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/github.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  renderTwitterButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.TwitterStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/twitter.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  renderAmazonButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.AmazonStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/amazon.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  renderMicrosoftButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.MicrosoftStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/windows.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  renderLinkedinButton(buttonText, opacity, height, width) {
    return (
      <TouchableOpacity
        style={[styles.LinkdnStyle, { height, width }]}
        activeOpacity={opacity}
        onPress={this.props.action}
        testID={this.props.testID}
      >
        <Image
          source={require("./images/linkedin.png")}
          style={styles.ImageIconStyle}
        />
        <View style={styles.SeparatorLine} />
        <Text style={styles.TextStyle}> {buttonText} </Text>
      </TouchableOpacity>
    );
  }

  render() {
    if (this.props.type == "facebook") {
      return (
        <View>
          {this.renderFaceBookButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    } else if (this.props.type == "amazon") {
      return (
        <View>
          {this.renderAmazonButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    } else if (this.props.type == "linkedin") {
      return (
        <View>
          {this.renderLinkedinButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    } else if (this.props.type == "google") {
      return (
        <View>
          {this.renderGoogleButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    } else if (this.props.type == "microsoft") {
      return (
        <View>
          {this.renderMicrosoftButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    } else if (this.props.type == "github") {
      return (
        <View>
          {this.renderGithubButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    } else {
      return (
        <View>
          {this.renderTwitterButton(
            this.props.text,
            this.props.opacity,
            this.props.height,
            this.props.width
          )}
        </View>
      );
    }
  }
}

const styles = StyleSheet.create({
  MainContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    margin: 10
  },
  GooglePlusStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#dc4e41",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  AmazonStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#FF9B00",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  GithubStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#333",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  FacebookStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#485a96",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  LinkdnStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#4875B4",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  InstagramStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#4E433C",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  TwitterStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#00acee",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  MicrosoftStyle: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#0078d7",
    borderWidth: 0.5,
    borderColor: "#fff",
    height: 40,
    width: 220,
    borderRadius: 5,
    margin: 5
  },
  ImageIconStyle: {
    padding: 10,
    margin: 5,
    height: 25,
    width: 25,
    resizeMode: "stretch"
  },
  TextStyle: {
    color: "#fff",
    marginBottom: 4,
    marginRight: 20
  },
  SeparatorLine: {
    backgroundColor: "#fff",
    width: 1,
    height: 40
  }
});
