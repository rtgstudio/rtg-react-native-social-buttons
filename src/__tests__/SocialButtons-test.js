import * as React from "react";
import { render } from "@testing-library/react-native";
import each from "jest-each";

import SocialButtons from "../SocialButtons";

describe("SocialButtons", () => {
  it(`renders the basics`, async () => {
    const renderer = render(<SocialButtons type="facebook" testID="test-id" />);
    await renderer.getByTestId("test-id");
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  each([
    "facebook",
    "amazon",
    "linkedin",
    "google",
    "microsoft",
    "github",
    "twitter"
  ]).test(`render the button for %s`, async type => {
    const renderer = render(
      <SocialButtons type={type} testID={`${type}-button`} />
    );
    await renderer.getByTestId(`${type}-button`);
  });
});
