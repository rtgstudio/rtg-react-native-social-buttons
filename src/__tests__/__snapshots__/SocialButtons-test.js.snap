// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtons render the login for amazon 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF9B00",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="amazon-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/amazon.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons render the login for facebook 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#485a96",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="facebook-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/facebook.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons render the login for github 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="github-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/github.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons render the login for google 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dc4e41",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="google-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/google-plus.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons render the login for linkedin 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4875B4",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="linkedin-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/linkedin.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons render the login for microsoft 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0078d7",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="microsoft-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/windows.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons render the login for twitter 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00acee",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="twitter-button"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/twitter.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;

exports[`SocialButtons renders the basics 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#485a96",
        "borderColor": "#fff",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": undefined,
        "margin": 5,
        "opacity": 1,
        "width": undefined,
      }
    }
    testID="test-id"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/facebook.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 5,
          "padding": 10,
          "resizeMode": "stretch",
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "height": 40,
          "width": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "marginBottom": 4,
          "marginRight": 20,
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;
